apidef/migration.go

```diff
@@ -13,30 +15,45 @@ func (a *APIDefinition) MigrateVersioning() (versions []APIDefinition, err error
 	}
 
 	if a.VersionData.NotVersioned && len(a.VersionData.Versions) > 1 {
-		return nil, errors.New("not migratable - if not versioned, there should be 1 version info in versions map")
+		return nil, errors.New("not migratable - if not versioned, there should be just one version info in versions map")
 	}
 
 	if !a.VersionData.NotVersioned {
+		if _, ok := a.VersionData.Versions[a.VersionData.DefaultVersion]; !ok {
+			return nil, errors.New("not migratable - if versioned, default version should match one of the versions")
+		}
+
 		for vName, vInfo := range a.VersionData.Versions {
 			if a.VersionData.DefaultVersion == vName {
 				continue
 			}
 
 			newAPI := *a
-			newAPI.APIID = ""
+
+			newID := uuid.NewV4()
+			apiID := strings.Replace(newID.String(), "-", "", -1)
+
+			newAPI.APIID = apiID
 			newAPI.Id = ""
+			newAPI.Name += "-" + url.QueryEscape(vName)
```

Comment:
For better readability can we use
```suggestion
			fmt.Sprintf("%s-%s", newAPI.Name, url.QueryEscape(vName)
```

Reply:
If the name wouldn’t have a value already I would agree. However, we are doing an append so I think it is more readable that we  add a suffix instead of replacement.